odxtools/diagcodedtypes.py

```diff
@@ -519,14 +551,14 @@ def convert_bytes_to_internal(self, decode_state: DecodeState, bit_position: int
             # if isinstance(param_value.parameter, LengthKeyParameter) would be prettier,
             # but leads to cyclic import...
             if (parameter.parameter_type == "LENGTH-KEY" and
-                    parameter.odx_id == self.length_key_id  # type: ignore
+                    parameter.odx_id == self.length_key.odx_id  # type: ignore
                ):
                 # The bit length of the parameter to be extracted is given by the length key.
                 assert isinstance(value, int)
                 bit_length = value
                 break
 
-        assert bit_length is not None, f"Did not find any length key with ID {self.length_key_id}"
+        assert bit_length is not None, f"Did not find any length key with ID {self.length_key.odx_id}"
```

Comment:
raise odx exception instead of assert

Reply:
I'd prefer not to open this can of worm in this PR. (I guess I should do the `assert` -> `odxassert` soon.)